examples/3-adapter_advanced.ipynb

```diff
@@ -0,0 +1,740 @@
+    "1. It gathers results from preceeding nodes.\n",
+    "2. It builds a dictionary with the same keys as the adapter and with values built according to descriptions:\n",
+    "   - if the key in the adapter maps to a single brick description, an appropriate object is extracted from the results of input nodes,\n",
+    "   - in the other case, objects are extracted according to brick descriptions and added to a list, which is then passed to a function that generates final object.\n",
+    "3. Arguments of `fit_transform` and `transform` are filled using the above dictionary."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's check if our ensembling works."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ensemble_step.fit_transform(data_train)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Looks fine! However, often we are interested only in the class with the highest probability. Let's make a step that will find this class for us."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "guesses_step = Step(name='guesses_maker',\n",
+    "                 transformer=Dummy(),\n",
+    "                 input_steps=[ensemble_step],                 \n",
+    "                 adapter={'y_pred': ([(ensemble_step.name, 'y_proba')],\n",
+    "                                     lambda lst: np.argmax(lst[0], axis=1))\n",
+    "                         },\n",
+    "                 cache_dirpath=CACHE_DIR,\n",
+    "                 force_fitting=True\n",
+    "                )"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "guesses_step"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You should be already familiar with everything that happened here. New step, `guesses_maker`, takes its input from `ensembler`. Adapter will create just one element: `y_pred`. List of bricks used to build `y_pred` has only one element:  `y_proba` found in `ensembler`'s result. Function `lambda lst: np.argmax(lst[0], axis=1)` takes this list and performs row-wise `argmax` on its only element."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "guesses_step.fit_transform(data_train)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In one of the previous cells we checked quality of our model manually. Let's add a final step that will do it for us automatically!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class EvaluationTransformer(BaseTransformer):\n",
```

Comment:
Hmm... making a step that does evaluation is very clever but It makes me really confused. @jakubczakon and @kamil-kaczmarek, what do you think?

Reply:
@mromaniukcdl @kamil-kaczmarek @grzes314 We were discussing it with @kamil-kaczmarek a while back. I have to say that I like to have pipeline (model) and evaluation as two separate concepts. It bears less risk of data leakage, You can build various pipelines and evaluate them with the same evaluation scheme. However using steps to do everything from fetching data from db and setting up the CV scheme all the way through producing reports and sending an email to the manager is not out of the picture here. I would distinguish however, pipelines from workflows where the first is pure modeling, second however does everything. Notice that pipeline can easily be just a subset of the workflow.